pysteps/verification/salscores.py

```diff
@@ -0,0 +1,386 @@
+
+    Parameters
+    ----------
+    prediction: array-like
+        Array of shape (m,n) with prediction data. NaNs are ignored.
+    observation: array-like
+        Array of shape (m,n)  with observation data. NaNs are ignored.
+    thr_factor: float
+        Factor used to compute the detection threshold as in eq. 1 of :cite:`WHZ2009`.
+        If specified, this is used to identify coherent objects enclosed by the threshold
+        contour thr_factor * quantile95(precip).
+    tstorm_kwargs: dict
+        Optional dictionary containing keyword arguments for the tstorm feature
+        detection algorithm. If None, default values are used.
+        See the documentation of :py:func:`pysteps.feature.tstorm.detection`.
+
+    Returns
+    -------
+    location_2: float
+        The secibd parameter of location component with value between 0 to 1.
+    """
+    maximum_distance = sqrt(
+        ((observation.shape[0]) ** 2) + ((observation.shape[1]) ** 2)
+    )
+    obs_r = (_sal_weighted_distance(observation, thr_factor, tstorm_kwargs)) * (
+        np.nanmean(observation)
+    )
+    forc_r = (_sal_weighted_distance(prediction, thr_factor, tstorm_kwargs)) * (
+        np.nanmean(prediction)
+    )
+    location_2 = 2 * ((abs(obs_r - forc_r)) / maximum_distance)
+    return float(location_2)
+
+
+def _sal_detect_objects(precip, thr_factor, tstorm_kwargs):
+    """This function detects thunderstorms using a multi-threshold approach from :cite:`Feldmann2021`.
+
+    Parameters
+    ----------
+    precip: array-like
+        Array of shape (m,n) containing input data. Nan values are ignored.
+    thr_factor: float
+        Factor used to compute the detection threshold as in eq. 1 of :cite:`WHZ2009`.
+        If specified, this is used to identify coherent objects enclosed by the threshold
+        contour thr_factor * quantile95(precip).
+    tstorm_kwargs: dict
+        Optional dictionary containing keyword arguments for the tstorm feature
+        detection algorithm. If None, default values are used.
+        See the documentation of :py:func:`pysteps.feature.tstorm.detection`.
+
+    Returns
+    -------
+    precip_objects: pd.DataFrame
+        Dataframe containing all detected cells and their respective properties.
+    """
+    if not PANDAS_IMPORTED:
+        raise MissingOptionalDependency(
+            "The pandas package is required for the SAL "
+            "verification method but it is not installed"
+        )
+    if not SKIMAGE_IMPORTED:
+        raise MissingOptionalDependency(
+            "The scikit-image package is required for the SAL "
+            "verification method but it is not installed"
+        )
+    if tstorm_kwargs is None:
+        tstorm_kwargs = dict()
+    if thr_factor is not None:
+        zero_value = np.nanmin(precip)
+        tstorm_kwargs = {
+            "minsize": tstorm_kwargs.get("minsize", 5),
+            "minmax": tstorm_kwargs.get("minmax", 0),
```

Comment:
as minmax is a threshold that applies to the field exceeding minref, I would suggest to put minmax=minref to disable it
"0" can be a problem, if minref happens to be negative

Reply:
done in 721b23a, thanks!